bundles/org.openhab.binding.upnpcontrol/src/main/java/org/openhab/binding/upnpcontrol/internal/handler/UpnpRendererHandler.java

```diff
@@ -244,11 +246,23 @@ public void stop() {
      * Invoke Play on UPnP AV Transport.
      */
     public void play() {
-        Map<String, String> inputs = new HashMap<>();
-        inputs.put("InstanceID", Integer.toString(avTransportId));
-        inputs.put("Speed", "1");
+        // wait for maximum 1s until the media URI is set before playing
+        for (int i = 0; isSettingURI && (i < 4); i++) {
+            try {
+                Thread.sleep(250);
+            } catch (InterruptedException e) {
+                continue;
```

Comment:
Do you mean `break`?

Reply:
Yes. But I now solved it differently with a CompletableFuture.